### ISSUE TITLE
tf.reduce_min([inf, nan]).eval() == inf, should be nan

### ISSUE THREAD

**Issue by girving** [tensorflow/tensorflow#17056]
### Describe the problem

`tf.reduce_min` and `tf.reduce_max` do not propagate nans correctly.  E.g.,

    >>> tf.reduce_min([inf, nan]).eval()
    inf

but the correct answer (the one most useful for debugging) is `nan`.  This is presumably due to a backwards comparison, similar to the old `tf.nn.relu(nan) == 0` bug that @alexalemi found.

### Source code / logs
Here's a colab illustrating the problem with TensorFlow 1.6.0-rc1: https://drive.google.com/file/d/1nDA0Q48PveBlx_D5Zurchbw8l7eczRSB/view?usp=sharing

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
Have I written custom code
OS Platform and Distribution
TensorFlow installed from
TensorFlow version
Bazel version
CUDA/cuDNN version
GPU model and memory
Exact command to reproduce

**Comment by girving** [comment]
OS Platform and distribution: colab
TensorFlow installed from: intrinsic to colab
TensorFlow version: 1.6.0-rc1
Bazel version: n/a
CUDA/cuDNN version: none
GPU model and memory: none
Exact command to reproduce: `tf.reduce_min([inf, nan]).eval()` or run the colab

**Comment by girving** [comment]
Oops, looks like the colab was only shared within OpenAI.  Should be fixed.

**Comment by girving** [comment]
Related issue: https://github.com/tensorflow/tensorflow/issues/12659

**Comment by angersson** [comment]
@mrry Is the underlying issue a dupe of #12659?

**Comment by mrry** [comment]
It sounds like it. Closing as a duplicate of #12659.

**Comment by girving** [comment]
@mrry A dupe means that fixing #12659 will necessarily fix this one, which isn't obvious to me.  Similar causes doesn't mean duplicate.

**Comment by mrry** [comment]
Assigning to @benoitsteiner, since he's got #12569, and is best placed to figure out if there is a common solution.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @benoitsteiner: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @benoitsteiner: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @benoitsteiner: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @benoitsteiner: It has been 15 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @benoitsteiner: It has been 78 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @benoitsteiner: It has been 95 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @benoitsteiner: It has been 110 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @benoitsteiner: It has been 125 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @benoitsteiner: It has been 140 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @benoitsteiner: It has been 155 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @benoitsteiner: It has been 170 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by girving** [comment]
Gah, this is a bug in `fmin` itself: https://en.cppreference.com/w/c/numeric/math/fmin :(

**Comment by girving** [comment]
Julia thread which similarly concluded that the C standard is broken and fixed Julia to not follow the broken C standard: https://github.com/JuliaLang/julia/issues/7866

**Comment by tensorflowbutler** [comment]
Nagging Assignee @benoitsteiner: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by mrry** [comment]
We depend on  `Eigen::internal::MinReducer<float>()` for the implementation of `tf.reduce_min()`, so I'm assigning this to @rmlarsen, who's most qualified to comment on the Eigen behavior.

**Comment by facaiy** [comment]
Is it related to #12659 ?

**Comment by tensorflowbutler** [comment]
Nagging Assignee @rmlarsen: It has been 15 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @rmlarsen: It has been 30 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @rmlarsen: It has been 45 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @rmlarsen: It has been 61 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @rmlarsen: It has been 76 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @rmlarsen: It has been 91 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by mohantym** [comment]
Hi @girving ! This issue is not replicating in TF [2.7](https://colab.sandbox.google.com/gist/mohantym/9b256232072da5f2e830d54bb1085fe1/github_17056.ipynb#scrollTo=iwNRaKeRrK8M) .Closing as it is resolved in latest version. Thanks!

**Comment by mohantym** [comment]
Are you satisfied with the resolution of your issue?
[Yes](https://goo.gl/forms/Oe0tEvODFRoI2gJF3)
[No](https://goo.gl/forms/fUjzOfrtkFbrOT8d2)
